drivers/display/display_dummy.c

```diff
@@ -24,50 +24,50 @@ static int dummy_display_init(struct device *dev)
 	return 0;
 }
 
-static int dummy_display_write(const struct device *dev, const uint16_t x,
+static int dummy_display_write(struct device *dev, const uint16_t x,
```

Comment:
This change is also problematic as an API change.  I tried removing the commit and the sample still builds.

There's work in progress to make all `struct device` instances const qualified "soon", so this change would be reverted when that happens.  It'd be better to not do this if you can avoid it.  Can you eliminate the warnings by adding casts inside the shim implementation (if you need a non-const driver reference)?

Reply:
done